drivers/pinmux/stm32/pinmux_stm32l4x.h

```diff
@@ -72,11 +72,15 @@
 
 #define STM32L4X_PINMUX_FUNC_PA9_USART1_TX                                     \
 	(STM32_PINMUX_ALT_FUNC_7 | STM32_PUSHPULL_PULLUP)
+#define STM32L4X_PINMUX_FUNC_PA9_I2C1_SCL                                      \
+	(STM32_PINMUX_ALT_FUNC_4 | STM32_OPENDRAIN_PULLUP)
```

Comment:
might be specific to some stm32l4xx soc devices.
can you check PA9/PA10 alternate function AF4 on L4 soc series, for example stm32l476 has no I2C1 pins on Port A

Reply:
The status of the L4 series I have checked is as follows.

|                 | L432 | L433 | L452 | L471 | L475 | L476 | L496 | L4r5,r9 |
| --------------- | ---- | ---- | ---- | ---- | ---- | ---- | ---- | ------- |
| PA9/PA10(I2C_1) | O    | O    | O    | X    | X    | X    | X    | X       |
| PB8/PB9(CAN)    | X    | O    | O    | O    | O    | O    | O    | O       |
| PD0/PD1(CAN)    | X    | O    | O    | O    | O    | O    | O    | O       |

I have applied the above to the source.

Reply:
However for stm32l476, that's not what is in the  [DS10198](https://www.st.com/content/ccc/resource/technical/document/datasheet/c5/ed/2f/60/aa/79/42/0b/DM00108832.pdf/files/DM00108832.pdf/jcr:content/translations/en.DM00108832.pdf)

Reply:
Sorry. I don't understand well.
The following code does not meet the conditions in `stm32l476`?
``` c
#if CONFIG_SOC_STM32L432XX || CONFIG_SOC_STM32L433XX || CONFIG_SOC_STM32L452XX
#define STM32L4X_PINMUX_FUNC_PA9_I2C1_SCL                                      \
	(STM32_PINMUX_ALT_FUNC_4 | STM32_OPENDRAIN_PULLUP)
#endif

...
...

#if CONFIG_SOC_STM32L432XX || CONFIG_SOC_STM32L433XX || CONFIG_SOC_STM32L452XX
#define STM32L4X_PINMUX_FUNC_PA10_I2C1_SDA                                     \
	(STM32_PINMUX_ALT_FUNC_4 | STM32_OPENDRAIN_PULLUP)
#endif
```

Reply:
yes, this code is _not_ valid  for stm32L476  (according to DS10198 Rev 8)

-  Alternate Function 4 for PA9 and PA10 is mapping I2C1 SCL and SDA for stm32l423, 433, 452 (but not stm32l476)

Reply:
`STM32L4X_PINMUX_FUNC_PA9_I2C1_SCL` is defined only when `CONFIG_SOC_STM32L432XX`, `CONFIG_SOC_STM32L433XX`, `CONFIG_SOC_STM32L452XX` are defined.
Isn't this compatible with stm32L476?

Reply:
correct: "STM32L4X_PINMUX_FUNC_PA9_I2C1_SCL is defined only when CONFIG_SOC_STM32L432XX, CONFIG_SOC_STM32L433XX, CONFIG_SOC_STM32L452XX are defined."
Maybe others but NOT stm32L476.